cmd/config_printer/config_summary.go

```diff
@@ -0,0 +1,103 @@
+/***************************************************************
+ *
+ * Copyright (C) 2024, Pelican Project, Morgridge Institute for Research
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ***************************************************************/
+
+package config_printer
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/pelicanplatform/pelican/config"
+)
+
+func configSummary(cmd *cobra.Command, args []string) {
```

Comment:
Right now this command doesn't work for me:
```
# pelican config summary
DEBUG[2024-11-04T17:08:20Z] Configured federation URL:                   
DEBUG[2024-11-04T17:08:20Z] Federation URL is unset; skipping discovery  
DEBUG[2024-11-04T17:08:20Z] Setting global jwks url to                   
panic: reflect: call of reflect.Value.IsNil on zero Value

goroutine 1 [running]:
reflect.Value.IsNil(...)
        /usr/local/go/src/reflect/value.go:1568
github.com/pelicanplatform/pelican/cmd/config_printer.compareStructsAsym({0x13eec40, 0x400000db90}, {0x0, 0x0?})
        /workspaces/pelican/pelican/cmd/config_printer/config_summary.go:90 +0x460
github.com/pelicanplatform/pelican/cmd/config_printer.compareStructsAsym({0x16d93c0, 0x4000f846c0}, {0x16d93c0, 0x4000f84900?})
        /workspaces/pelican/pelican/cmd/config_printer/config_summary.go:79 +0x5ac
github.com/pelicanplatform/pelican/cmd/config_printer.compareStructsAsym({0x13c1940, 0x40000bb100}, {0x13c1940, 0x40000b9600?})
        /workspaces/pelican/pelican/cmd/config_printer/config_summary.go:79 +0x5ac
github.com/pelicanplatform/pelican/cmd/config_printer.configSummary(0x4000e90a00?, {0x16e874a?, 0x4?, 0x16e870e?})
        /workspaces/pelican/pelican/cmd/config_printer/config_summary.go:43 +0xa4
github.com/spf13/cobra.(*Command).execute(0x3430bc0, {0x3498be0, 0x0, 0x0})
        /root/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x654
github.com/spf13/cobra.(*Command).ExecuteC(0x3430040)
        /root/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x320
github.com/spf13/cobra.(*Command).Execute(...)
        /root/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /root/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:985
main.Execute()
        /workspaces/pelican/pelican/cmd/root.go:90 +0x94
main.handleCLI({0x40000a61e0, 0x3, 0x3})
        /workspaces/pelican/pelican/cmd/main.go:66 +0x1dc
main.main()
        /workspaces/pelican/pelican/cmd/main.go:35 +0x34
```

Reply:
Addressed it in the commit: https://github.com/PelicanPlatform/pelican/pull/1693/commits/c17bc61b341e9076f46c02382b221665211a4d96